### ISSUE TITLE
[RFC] Screener visual test for all themes

### ISSUE THREAD

**Issue by bmdalex** [microsoft/fluent-ui-react#992]
# Screener visual test for all themes

### Problem description

We need visual tests for the other themes Stardust supports

### Solution 1: add URL parameter to maximized examples

Following the approach in #792 on adding support for RTL examples, we can add a URL parameter for the theme name. The route (see: https://github.com/stardust-ui/react/blob/35ab4603198e2f373f27e1bcdba11c9996a0ffc1/docs/src/routes.tsx#L21) will become something like:

```jsx
<Route
  exact
  path="/maximize/:exampleName/:themeName/:rtl?"
  component={ExternalExampleLayout}
/>
```

### -:
- what I'm concerned about is that because we will add this for 2 other themes, the amount of time for tests will triple from **6 min to 18 min**

### Solution 2: render components in all themes in maximized examples

An alternative approach I was considering was to change `docs/src/components/ExternalExampleLayout.tsx` to render the exampleSource in all 3 themes in the same file;

### +:
- no extra time for tests, **currently it's about 6min and it'd become 18min..** - I prefer this approach for this reason alone
- simplicity (technically speaking)

### -:
- tests will fail for visual differences in any theme (still possible to understand from screenshots where it happened)
- we need to change the selectors and approach for tests with states as they expect only one component to be rendered, so we'd have to rewrite the steps so that they apply for the components in other themes as well

_from @layershifter:_
#### `Loader.steps.ts`

```tsx
import { themes } from '@stardust-ui/react'

const steps = [
  steps => steps.switchTheme(teams.dark).snapshot('Loader in Dark theme'),
  steps => steps.switchTheme(teams.hc).snapshot('Loader in HC theme'),
]
```

Technically we can implement this in `ExternalExampleLayout` as it was done for examples reset.

### +:
- we will add only 2 more snapshots for an example
- we will not need to write specific examples for each theme

### -:
- more verbose: we will need to go to every flavor of every component and write a file to call the function to do the theme switching
- adds to total time for tests but not as much as 1st approach

**Comment by bmdalex** [comment]
fyi @layershifter and @mnajdova if you want to paste your comments here

**Comment by layershifter** [comment]
### Option 3: extend `Steps` API

#### `Loader.steps.ts`

```tsx
import { themes } from '@stardust-ui/react'

const steps = [
  steps => steps.switchTheme(teams.dark).snapshot('Loader in Dark theme'),
  steps => steps.switchTheme(teams.hc).snapshot('Loader in HC theme'),
]
```

- we will add only 2 more snapshots for an example
- we will not need to write specific examples for each theme

Technically we can implement this in `ExternalExampleLayout` as it was done for examples reset.

**Comment by mnajdova** [comment]
In the end after thinking about this, I agree with @layershifter just let's make sure that we can make this reusable, and maybe just use some service like `createStepsForThemes` or something similar.. Because if we decide to add the `base` theme later for example, we won't have to do too much changes.

**Comment by bmdalex** [comment]
I'm inclining towards solution no. 3 as well

**Comment by miroslavstastny** [comment]
I am against having multiple examples in a single screenshot, it makes comparisons more difficult.
I like Option 3 but do it automatically so that we do not need to add any code to all examples (just add some flag at most)

**Comment by bmdalex** [comment]
@miroslavstastny I'm all if for minimizing the amount of code we write. However, I fail to see where we can add this flag and how to implement that. I'll investigate more and come back to this, but what are you suggesting?

I'm thinking:
- change the file extensions for components we want to test in other themes (same as we did for rtl)
- add some prop to `ComponentExample` component and import the steps for switching themes based on that (if we choose this we should change rtl approach as well)

@layershifter @mnajdova

**Comment by layershifter** [comment]
Was implemented in #1043.
